### PR TITLE
getAlive and getDead functions

### PR DESCRIPTION
**Pull request by TechnichalDoge**
These need to be tested and checked over, but these adds getAlive() and getDead() functions which it's function is on the tin.
This adds in a shorter way and simpler way for e2ers to get all the people are alive or dead, which is useful for a many a reason.

### CODE REVIEW

File: lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua [hunk 1]
```diff
@@ -103,3 +103,25 @@ e2function number maxLoss()
 
     return max
 end
+
+e2function array getAlive()
+    local arrayOfLiving = {}
+    local humans = player.GetHumans()
+
+    for _, human in pairs( humans ) do
+        if human:Alive() then arrayOfLiving[#arrayOfLiving + 1] = human end
```
Comment (OverlordBleck) [inline lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua]: Can you use the `table.insert` method instead, #ARRAY tends to give false readings. In this case, it shouldn't be an issue but better safe than sorry.

File: lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua [hunk 2]
```diff
@@ -103,3 +103,25 @@ e2function number maxLoss()
...
+
+    return arrayOfLiving
+end
+
+e2function array getDead()
+    local arrayOfDead = {}
+    local humans = player.GetHumans()
+
+    for _, human in pairs( humans ) do
+        if not human:Alive() then arrayOfDead[#arrayOfDead + 1] = human end
```
Comment (OverlordBleck) [inline lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua]: The same goes for here.

File: lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua (hunk 1 above)
Comment (OverlordBleck) [reply lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua]: Also, I'm not that big of a fan of the one line if statement in this case... Maybe @brandonsturgeon  could give a better way of doing it right.

**Comment by TechnichalDoge** [comment]
Does that do?

**Comment by TechnichalDoge** [comment]
I dont think these functions belong under "array"

File: lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua [hunk 3]
```diff
@@ -103,3 +103,25 @@ e2function number maxLoss()
 
     return max
 end
+
+e2function array getAlive()
+    local arrayOfLiving = {}
+    local humans = player.GetHumans()
+
+    for _, human in pairs( humans ) do
+        if human:Alive() then table.insert(arrayOfLiving, human) end
```
Comment (OverlordBleck) [inline lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua]:
```suggestion
        if human:Alive() then table.insert( arrayOfLiving, human ) end
```

File: lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua [hunk 4]
```diff
@@ -103,3 +103,25 @@ e2function number maxLoss()
...
+
+    return arrayOfLiving
+end
+
+e2function array getDead()
+    local arrayOfDead = {}
+    local humans = player.GetHumans()
+
+    for _, human in pairs( humans ) do
+        if not human:Alive() then table.insert(arrayOfDead, human) end
```
Comment (OverlordBleck) [inline lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua]:
```suggestion
        if not human:Alive() then table.insert( arrayOfDead, human ) end
```

**Comment by OverlordBleck** [comment]
> I dont think these functions belong under "array"

Either way, it's going to need a helper message. I'll trust you to put it where you think is most fit.

**Comment by TechnichalDoge** [comment]
Does that do.

File: lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua (hunk 1 above)
Comment (PLally) [reply lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua]: #array is fine it only gives "false readings" if you don't know what you're doing and you use it for non sequential arrays. in cases were you are constructing the array yourself i don't see a problem with using it.

Comment (TechnichalDoge) [reply lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua]: what about the one line if statement bleck is iffy about?

Comment (PLally) [reply lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua]: the question is which one is more efficient

Comment (TechnichalDoge) [reply lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua]: I'm not sure if it's feasible to try to test that.

Comment (brandonsturgeon) [reply lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua]: Definitely go with the `insert`

Comment (TechnichalDoge) [reply lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua]: what about how bleck said he was iffy about the one liner

Comment (OverlordBleck) [reply lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua]: I just thought it looked ugly that's all...

Comment (brandonsturgeon) [reply lua/entities/gmod_wire_expression2/core/custom/cfc_methods.lua]:
```lua
        if human:Alive() then table.insert( arrayofLiving, human) end
```
Looks okay to me.

One could make the argument that this was acceptable, too:
```lua
        local function addHuman( human)
            table.insert( arrayOfLiving, human )
        end

        if human:Alive() then addHuman( human ) end
```

**Comment by TechnichalDoge** [comment]
Need final comments or whatnot

**Comment by samuelWilliams99** [comment]
Tested, all seems to work fine, missing op costs, but so is many other functions in here, will raise an issue.

**Pull request merged**